### PR TITLE
reduce inflatino when close to goal + more features

### PR DESCRIPTION
**Pull request by younesr1**
This PR does a couple of things:
1. This PR reduces the costmap inflation when the robot is < 2.5 meters away from the goal and inflates it back up when its more than 7 meters from the goal. this allows the robot to get up real close to the bed.
2. This PR increases the planner patience. The planner used to wait 5 seconds before beginning space clearing behaviors. it now waits 15 seconds. 
3. it increase sthe shape of the polygon the base uses during nav bc we now drive with a box
4. it moves rotational space clearing behavior until after an aggressive reset of the costmap is performed. the rotational clearing is desperate and should be done after the aggressive map reset. i also added another aggressive map reset after the rotational clearing
5. we now clear the costmaps on every navigation request. on thursday, i noticed that the robot would last see someone behind a hallway before it makes a turn and never sees them leave. so it assumes that person is still there which causes it to be unable to return home. clearing the costmap should leave only static obstacles and currently observable obstacles (in theory. i will test later)

### CODE REVIEW

File: nightingale_navigation/configs/costmap_common_params.yaml [hunk 1]
```diff
@@ -15,5 +16,5 @@ inflation_radius: 0.45
 obstacle_layer/footprint_clearing_enabled: true
 
 #raytrace_range must be less than max range of sensor.
-observation_sources: scan_multi
-scan_multi: {data_type: LaserScan, topic: movo/scan_multi, marking: true, clearing: true, inf_is_valid: true, obstacle_range: 4.75, raytrace_range: 8.0}
+observation_sources: base_scan_filtered
```
Comment (FelixKHWong) [inline nightingale_navigation/configs/costmap_common_params.yaml:19]: What was the decision to change to base_scan_filtered?

**Review by FelixKHWong — DISMISSED** [review]
looks good. I'm assuming the behaviour was testing on Thursday because we were able to see the costmap clear and also get close to the bed?

File: nightingale_navigation/configs/costmap_common_params.yaml (hunk 1 above)
Comment (younesr1) [reply nightingale_navigation/configs/costmap_common_params.yaml:19]: the laser scanner kept reporting a flickering obstacle in the hallway when the robot was in the room. the obstacle was not real bc the robot was in the lab and cannot see into the hallway. If i had to guess, id say the culprit is the reflective wheel on the desk the robohub folks just added to the room. the presence of this fake flickering point caused us to take the "long way around" to the stocking area. it was transient so didnt happen every time. 

anyway, the filtered scan rejects this lone, flickering point.

Comment (younesr1) [reply nightingale_navigation/configs/costmap_common_params.yaml:19]: the flickering point exists in the unfiltered scan, but does not exist in the filtered scan

File: nightingale_navigation/src/room_runner.py [hunk 2]
```diff
@@ -147,6 +191,14 @@ def fb_cb(fb):
                 [fb.base_position.pose.position.x, fb.base_position.pose.position.y]
             )
             euclidean_distance = np.linalg.norm(target - cur)
+            if euclidean_distance < self.deflation_hysteresis_low:
+                self.dynamic_reconfigure_client.update_configuration(
+                    {"inflation_radius": self.local_inflation / 3}
```
Comment (le2tang) [inline nightingale_navigation/src/room_runner.py:196]: Maybe add a parameter to control the deflation factor

File: nightingale_navigation/src/room_runner.py [hunk 3]
```diff
@@ -147,6 +191,14 @@ def fb_cb(fb):
                 [fb.base_position.pose.position.x, fb.base_position.pose.position.y]
             )
             euclidean_distance = np.linalg.norm(target - cur)
+            if euclidean_distance < self.deflation_hysteresis_low:
```
Comment (le2tang) [inline nightingale_navigation/src/room_runner.py:194]: I'm not sure this line guarantees hysteresis. Is hysteresis the desired behaviour? It seems like this method might work fine

**Review by le2tang — DISMISSED** [review]

File: nightingale_navigation/src/room_runner.py (hunk 3 above)
Comment (nightingale-project) [reply nightingale_navigation/src/room_runner.py:194]: im not sure what u mean?

Comment (nightingale-project) [reply nightingale_navigation/src/room_runner.py:194]: yes, hysteresis is desired

**Pull request merged**